src/r2x/exporter/sienna.py

```diff
@@ -432,3 +428,127 @@ def export_data(self) -> None:
         # First export all time series objects
         self.export_data_files()
         logger.info("Saving time series data.")
+
+
+def apply_operation_table_data(
+    component: dict[str, Any],
+) -> dict[str, Any]:
+    """Process and apply operation cost data for `PSY.power_system_table_data.jl`.
+
+    This function extracts operation cost data from a component dictionary and adds
+    various cost-related fields to it. It handles different types of cost data,
+    including variable costs, fuel costs, heat rates, and cost/fuel curves.
+
+    Parameters
+    ----------
+    component : dict[str, Any]
+        A dictionary containing component data, potentially including operation cost information.
+
+    Returns
+    -------
+    dict[str, Any]
+        The input component dictionary, potentially modified with additional operation cost fields.
+
+    Notes
+    -----
+    This function modifies the input dictionary in-place and returns it.
+
+    The function processes the following types of operation costs:
+    - Variable costs (vom_cost)
+    - Fuel costs
+    - Heat rate coefficients (a0, a1, a2)
+    - Cost curves
+    - Fuel curves
+
+    Raises
+    ------
+    AssertionError
+        If the fuel_cost is None when present in the variable cost data.
+    NotImplementedError
+        If an unsupported variable curve type is encountered.
+
+    Examples
+    --------
+    >>> component = {
+    ...     "operation_cost": {
+    ...         "variable": {
+    ...             "vom_cost": {"function_data": {"proportional_term": 10}},
+    ...             "fuel_cost": 0.05,
+    ...             "value_curve": {
+    ...                 "function_data": {
+    ...                     "constant_term": 100,
+    ...                     "proportional_term": 20,
+    ...                     "quadratic_term": 0.5,
```

Comment:
small thing but when would you have quadratic and PPC terms within one value curve?